File: C++SrcFiles/ShardShoot.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cmath>
#include <algorithm>

using namespace std;

class ShardShoot : public MonoBehaviour
{
private:
    GameObject* shardPrefab;
    Transform* shardSpawn;
    float attackTimer;

public:
    ShardShoot()
    {
        shardPrefab = Resources::Load<GameObject>("Prefabs/Shard");
        shardSpawn = transform->Find("ShardSpawn");
        attackTimer = 0.0f;
    }

    void Update()
    {
        // Count up the attack timer
        attackTimer += Time.deltaTime;

        // Check if it's time to attack
        if (attackTimer >= 2.0f) {
            attackTimer = 0.0f;
            Shoot();
        }
    }

    void Shoot()
    {
        // Instantiate a shard object at the shard spawn position
        Instantiate(shardPrefab, shardSpawn->position, Quaternion::identity);
    }
};

File: C#SrcFiles/ShardShoot.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class ShardShoot : MonoBehaviour
{
    // the prefab of the reaperattack png;
    public GameObject Shard;
    //position of the attack;
    public Transform shardspawn;
    
    // Intervals between attacks;
    private float attacktimer;
    void Update()
    {
        // counting till 3 seconds and then shooting the reaper bullet.
        attacktimer += Time.deltaTime;
        if (attacktimer > 2)
        {
            attacktimer = 0;
            shoot();
        }
    }


    private void shoot()
    {
        Instantiate(Shard, shardspawn.transform.position, Quaternion.identity);
    }
}